### ISSUE TITLE
item too long to fit on line is stuck without any progress

### ISSUE THREAD

**Issue by sachin-suresh-rapyuta** [brechtm/rinohtype#351]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Sphinx or rinoh output

```text
item too long to fit on line (page 2671)      ] ETA 12:36 (10:50) page 2671
item too long to fit on line (page 2672)      ] ETA 12:36 (10:50) page 2672
item too long to fit on line (page 2673)      ] ETA 12:36 (10:50) page 2673
item too long to fit on line (page 2674)      ] ETA 12:37 (10:51) page 2674
item too long to fit on line (page 2675)      ] ETA 12:37 (10:51) page 2675
item too long to fit on line (page 2676)      ] ETA 12:37 (10:51) page 2676
item too long to fit on line (page 2677)      ] ETA 12:38 (10:52) page 2677
item too long to fit on line (page 2678)      ] ETA 12:38 (10:52) page 2678
item too long to fit on line (page 2679)      ] ETA 12:39 (10:53) page 2679
Killed
```

### Source files

In conf.py file, on using the below config, and when I run the command: `sphinx-build -b rinoh source build/rinoh`, the message - "item too long to fit on line" hangs indefinitely and doesn't progress. 

```
rinoh_documents = [dict(doc='index',        
                       target='pa-amr-specification-doc',
                        template='rinohtype.rtt',
                        logo='_static/rr-logo-vertical2022-1100px-transp.png')]
```

But, when I comment this code and run the command, it progresses, but taking a very long time to generate the PDF. I am still not sure that the document I had written is more than 3000 pages as this is what the output is showing (like: ETA 10:00 page 2985) for 30% progress.

### Versions

```text
$ sphinx-build --version
sphinx-build 4.5.0

$ python --version
Python 3.8.10
```

**Comment by brechtm** [comment]
Do you mean that the rendering ends up in an endless loop?

Note that, as the new issue form mentions, I cannot debug this without having access to your Sphinx project or a minimal example reproducing the issue.

**Comment by sachin-suresh-rapyuta** [comment]
Yes, as you can see from the output screenshot, the pdf generation is stuck at 14% and I don't know where or which .rst file it is pointing to to correct the "item too long" message.

**Comment by brechtm** [comment]
There are several things in your document that cause the rendering loop to get stuck:

1. You are running into bug #283, triggered by _fuzzy_pattern_without_border.png_ and other tall images. Scaling these down works around this problem:
   ```rst
   .. figure:: ./images/getting_started/map_creation_checklist/fuzzy_pattern_without_border.png
      :scale: 50%
      ...
   ```
   This is a high-priority bug, but I'm not sure when I'll be able to fix it.

2. rinohtype is not yet able to split table cells across two pages. This also leads to an infinite rendering loop when the contents of a cell do not fit on a page. For example, the _Comparison between Full-fledged WCS vs Minimalistic WCS_ table triggers this issue. Removing the `:widths:` option happens to make the contents fit. There are however other tables (often containing images) in the document that trigger this bug.

   This is also an important bug, but you should ask yourself whether tables are the best choice for that kind of content. Do you really want table cells that span multiple pages?

Some general remarks about your document that occurred to me when trying to debug this:
- URL links are created using raw-html substitutions which only work with HTML builders. I suppose you do this to make them open in another window/tab (`target="_blank"`)?
- The A5 paper size allows for very little room, causing many "item too long" warnings. As I already mentioned in https://github.com/brechtm/rinohtype/discussions/337#discussioncomment-3085246, you should consider decreasing the page margins.
